rfcs/2020-04-15-wasm-foreign-modules.md

```diff
@@ -0,0 +1,559 @@
+# RFC #2341 - 2020-04-15 - WASM Foreign Module Support
+
+Introduce Vector's foreign module support, a way for users to add custom functionality to Vector. This RFC introduces
+WASM modules specifically, but we envision the possibility of other implementations (these will be a different RFC).
+This RFC introduces Transforms and will be later amended with Source and Sink functionality.
+
+
+## Motivation
+
+There is a plurality of reasons for us to wish for flexible, practical foreign module support to Vector. These may also
+ be regarded as 'Plugins' or 'Extensions.' WASM is the most versatile and obvious choice for a first implementation as
```

Comment:
> for a first implementation

Is this proviso necessary? It implies that there are some other implementations of modules in plans, but doesn't mention it.

Reply:
I think you're right. :)